internal/trace/mock_tracer.go

```diff
@@ -37,6 +38,11 @@ type MockTracer struct {
 
 	// Sampled specifies if the new span should be sampled or not.
 	Sampled bool
+
+	// LastSpan is set with the name argument of each *MockTracer.Start()
```

Comment:
I think it would be better if we had `OnSpanStarted func(span *MockSpan)` that, if not nil, would be invoked every time a new `MockSpan` is created in `Start`. Then I'd add the `name` member and a `Name() string` getter to `MockSpan`. That way, the `TestSpanFormatter` function can install its own callback to get the created span and its name.

Reply:
@krnowak `*MockSpan` doesn't have the span name, should I add it as an exported field?

Reply:
I guess exported Name field is fine as well.

Reply:
@krnowak ops, I missed that you mentioned to add a `Name()` callback, happy to change what I have already if need be :)

Reply:
No no, it's fine that way too. I'd complain if this was sdk, not the mock stuff. :)